core.coffee

```diff
@@ -36,6 +36,59 @@
 		'Options without $ prefixes (e.g. `options: filter: ...`) are deprecated, please use a $ prefix (eg `options: $filter: ...`).'
 	)
 
+	class Poll
+		channels:
+			'error': []
+			'data': []
+
+		stopped: false
+
+		intervalTime: 10000
+
+		constructor: (requestFn, intervalTime) ->
+			if typeof intervalTime is 'number'
+				@intervalTime = intervalTime
+
+			@requestFn =  requestFn
+			@start()
+
+		runRequest: ->
+			return if @stopped
+			@requestFn()
+			.then (response) =>
+				return if @stopped
+
+				fn(response) for fn in @channels.data
+			.catch (err) =>
+				return if @stopped
+
+				fn(err) for fn in @channels.error
+
+		on: (name, fn) ->
+			@channels[name].push(fn)
+			return
+
+		start: ->
+			@stopped = false
+			@runRequest()
+			@pollInterval = setInterval =>
```

Comment:
It might be good to also offer the option to change the poll interval, so it's possible to change the poll interval depending on other factors (eg window focus)

Reply:
I've added a `setPollInterval` method